### ISSUE TITLE
Bug: Spell checker not working in french on Windows

### ISSUE THREAD

**Issue by mycanaletto** [ferdium/ferdium-app#86]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

v6.0.0-nightly.23

### What Operating System are you using?

Windows

### Operating System Version

10

### What arch are you using?

x64

### Expected Behavior

Thanks for the review of this fabulous app!

There is a problem with spell checker that affects the French. On a Windows 10 OS installed in English, whether the spell checker is configured in automatic or in French, the French disctionary does not install and the spell checking does not work.

Merci beaucoup

### Actual Behavior

It's ok in other languages

### Steps to reproduce

*

### Screenshots

![image](https://user-images.githubusercontent.com/45459269/166231758-7f294a03-d98c-4e99-bd8b-a16f8eef6053.png)

**Comment by vraravam** [comment]
@mycanaletto - Ferdium is not going to automatically install any language. The language dictionary has to be present outside of Ferdium - in which case, Ferdium can use it for spellchecking.

**Comment by mycanaletto** [comment]
I'm not sure. Spellchecking works very well in French outside of Ferdi (Chrome, Firefox) but not in Fermium. But this bug already existed in Ferdi. But it works for other languages than french, you can try...

**Comment by vraravam** [comment]
no - my question was around your screenshot: were you expecting Ferdium to download any dictionary? I don't think that's how it works. The French dictionary is considered a system/OS dependency - and should already be present for Ferdium to use.

**Comment by mycanaletto** [comment]
I think I remember that it can work with or without dictionaries. In automatic mode it is without and in manual mode of choice of language it must download the dictionary of the chosen language (exchanges on the old github). But whatever the combination and even if I add it by hand it does not change anything, the correction in French does not happen.

A French word well detected as well or badly written, but in this last case a right click proposes only words in English...

**Comment by vraravam** [comment]
<img width="877" alt="Screenshot 2022-05-03 at 9 42 36 AM" src="https://user-images.githubusercontent.com/69629/166475664-71aafa12-80ca-4cbb-86c0-636beadcbe04.png">
Right in this web page, I copy-pasted french text from translate.google.com - and it correctly recognized the word that was incorrectly spelt. But, this is on a mac - and if i remember correctly, for macos, Ferdium uses the in-built OS spellchecker.

**Comment by mycanaletto** [comment]
Yes, on the Mac it is the system checker, but not on Windows.

The fact is that it works in many languages, except French...

**Comment by SpecialAro** [comment]
![image](https://user-images.githubusercontent.com/37463445/168570364-69cf8554-112b-4c25-8809-f9b500f94f46.png)

If I understood this issue correctly, I'm unable to reproduce this issue on Windows 11 with 6.0.0-nightly.36

![image](https://user-images.githubusercontent.com/37463445/168570623-a8b2eb2e-5e47-4817-875c-9d2f0d5ff20a.png)

Maybe there is something wrong with your local @mycanaletto ?

**Comment by mycanaletto** [comment]
The word is marked as misspelled, but when you right click to choose a correction only words from the English dictionary are proposed.

This happens whether you are in automatic correction language detection or you choose French.

My Windows (10 or 11) are in English display language and the local FR.

This problem had already been reported on the githib ferdi by several people.

ps: I made a picture because if I make a screen shot I can't see the window opened by the right click.
![image](https://user-images.githubusercontent.com/45459269/168579019-30794317-e2a0-4780-9fcf-b5578cd8cc4a.png)

**Comment by mycanaletto** [comment]
No news on the beta 1, help !

**Comment by SpecialAro** [comment]
> The word is marked as misspelled, but when you right click to choose a correction only words from the English dictionary are proposed.

Ok! I think I finally understood what exactly you are saying.

In Portuguese, the word "exastamente" is wrong and should be "exatamente". And, as you can see in the figure bellow, the only word which is portuguese is "exatamente" but it suggests other words in english and also in other languages I think.

![image](https://user-images.githubusercontent.com/37463445/171446584-1a61ef1e-4ff2-47ba-87a9-845a9b5887dc.png)

We should probably look into it, but I don't have any idea how the dictionary is implemented in Ferdium neither if it is a Ferdium problem per se.

**Comment by mycanaletto** [comment]
Voilà ! Merci. I am of course ready to test when you start working for this problem ;-)

**Comment by kwilczynski** [comment]
Hello everyone!

Thank you for keeping the project alive!

I just moved over from Ferdi. Everything works fine aside of the spellchecking, where it does not work for me when I am using English. I am not sure why.

Ferdium was installed using Flatpak, as per:

```
Ferdium - Messenger for the desktop

          ID: org.ferdium.Ferdium
         Ref: app/org.ferdium.Ferdium/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 6.0.0b1
     License: Apache-2.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 342.0 MB
     Runtime: org.freedesktop.Platform/x86_64/21.08
         Sdk: org.freedesktop.Sdk/x86_64/21.08

      Commit: 1463c03fb8af3d110be40445cc0c98bc679ea8909b2f785769747eedcca2552e
     Subject: removed flathub.json (0f0f1b0f)
        Date: 2022-06-06 07:09:16 +0000
```

Thank you for help in advance!

**Comment by StephanBe** [comment]
I have to add, that there seems to be a bigger problem with the spell checker:

On my system (snap ferdium nightly 80 on Ubuntu 22.04) the spell checker is dead set on English and ignores all language settings.

(starting with e.g. ```LANG=de_DE.UTF-8 ferdium``` didn't help either)

**Comment by alanodx2** [comment]
Same problem. I just installed the latest version from AUR. Pt-Br dictionary doesn't work
Running on ArcoLinux

**Comment by SpecialAro** [comment]
Everyone that is facing this problem, can you please try setting a different language inside the recipe (Right click on the recipe icon, click Edit, scroll to Spell Checking Language) and change the "Use System Defaults"?

![image](https://user-images.githubusercontent.com/37463445/178857656-1b3d21f9-eb40-46a2-9f96-2483ede0da97.png)

The problem can be the overall setting not being correctly parsed to the recipe OR the `Use System Defaults` is faulty as it is.

**Comment by StephanBe** [comment]
> Everyone that is facing this problem, can you please try setting a different language inside the recipe (Right click on the recipe icon, click Edit, scroll to Spell Checking Language) and change the "Use System Defaults"?
> 
> The problem can be the overall setting not being correctly parsed to the recipe OR the `Use System Defaults` is faulty as it is.

Eureka! Setting the recipe spell checking language to the desired setting did solve the problem on my end. The global setting is indeed not mirrored to the recipe settings and more or less gets ignored completely.

Thanks for the workaround. @SpecialAro

**Comment by mycanaletto** [comment]
> 

You are lucky, it still doesn't work... (v 6.2 Win10/64)

**Comment by mycanaletto** [comment]
I just installed the last night and this problem persists...

**Comment by Darkein** [comment]
I have the same problem and found a wonderful workaround: I just copied a french dictionary from an other app, and replaced the english one 😄

**Comment by mycanaletto** [comment]
> I have the same problem and found a wonderful workaround: I just copied a french dictionary from an other app, and replaced the english one 😄

Yes, this way works. But it is not the most elegant one either ;-)

**Comment by nonobio** [comment]
Hi, I have the same problem with Ferdium 7.0.0 installed on Windows 10 French.

Most French words I write are underlined in red and correction proposals are not French:

![Image](https://github.com/user-attachments/assets/cb0c8da9-1129-459e-9b24-3172917fa4da)

My spelling verification adjustment is on auto-detect language:

![Image](https://github.com/user-attachments/assets/693caeb7-b35e-4d0d-b647-b352c2fc65b2)

I tried to pass it in French, to restart Ferdium but the problem is still present.

If I look in C:\Users\USER\AppData\Roaming\Ferdium\Dictionaries, the french bdic file isn't downloaded.
I chose other languages ​​in Ferdium like Deutch and Turkish, and after restarting Ferdium, the files are well downloaded:

![Image](https://github.com/user-attachments/assets/08db4fe9-b6c3-458a-bdec-d5da01870846)

While waiting for a solution, I downloaded [fr-FR-3-0.bdic](https://github.com/cvsuser-chromium/third_party_hunspell_dictionaries/blob/master/fr-FR-3-0.bdic) file, backed up the en-US-10-1.bdic file, deleted it, then renamed my French file to en-US-10-1.bdic. It works.

Thanks

**Comment by mycanaletto** [comment]
Yes, same problem here inn 2025 and th v 7.0

**Comment by mycanaletto** [comment]
Back to webCatalog....
